### PR TITLE
Add support for encoding of data and list based data

### PR DESCRIPTION
**Pull request by csanders-git**
Encoding of the HTTP message body is sometimes defined via the charset attribute within in the content-type header. Not all content-types allow charset, for instance multipart/form-data doesn't seem to. This is mostly there to address #64 but really UTF-8 is the default and pretty much the only one used. 

We also add the ability to read data values as a list. This list will be joined by CRLF and makes it way sexier to read multipart form submissions.

### CODE REVIEW

File: ftw/http.py [hunk 1]
```diff
@@ -335,12 +336,36 @@ def build_request(self):
...
             for hname, hvalue in self.request_object.headers.iteritems():
-                headers += unicode(hname) + ': ' + unicode(hvalue) + str(self.CRLF)
+                headers += unicode(hname) + ': ' + unicode(hvalue) + self.CRLF
         request = string.replace(request, '#headers#', headers)
 
         # If we have data append it
         if self.request_object.data != '':
-            data = str(self.request_object.data)
+            # Before we do that see if that is a charset
+            encoding = "utf-8"
```
Comment (zmallen) [inline ftw/http.py]: single quotes here

**Comment by zmallen** [comment]
can you add tests for the new functionality you added?

**Comment by csanders-git** [comment]
I added tests for this functionality and also added support for test_title instead of rule_id and fixed the conftest.py

File: ftw/http.py [hunk 2]
```diff
@@ -200,6 +200,7 @@ def process_response(self):
         self.response_line = response_line
         self.headers = response_headers
         self.data = response_data
+        print self.data
```
Comment (zmallen) [inline ftw/http.py]: remove print statements

File: ftw/http.py [hunk 3]
```diff
@@ -386,6 +387,8 @@ def build_request(self):
             request = request.decode('string_escape')
         # if we have an Encoded request we should use that
         self.request = request
+        print self.request
```
Comment (zmallen) [inline ftw/http.py]: remove print here

File: ftw/http.py [hunk 4]
```diff
@@ -386,6 +387,8 @@ def build_request(self):
...
+        print
```
Comment (zmallen) [inline ftw/http.py]: and here

**Comment by zmallen** [comment]
Going to come back to this, but I think if we name the tests a little more succinctly (aka where the testfile is) and you remove the print statements/fix some of the small nitpicks this should be good to merge

 I think that test_title is a good way to force the user to say what the test is about / where its located as before it was just rule_id, i just dont want to lose visibility into troubleshooting tests in the longrun

**Comment by zmallen** [comment]
LGTM

**Pull request merged**